### ISSUE TITLE
Margin Calculated Twice when Calculating Label Type Marginalia (on XYFrame)

### ISSUE THREAD

**Issue by torioLuz** [emeeks/semiotic#366]
It seems that margins are calculated twice, hence the available size to arrange labels become inaccurate.

The first one is already taken into account when calculating `adjustedSize` on [https://github.com/emeeks/semiotic/blob/master/src/components/XYFrame.js#L557](XYFrame.js)

`adjustedSize` is then used as `size` on `AnnotationLayer.js`, but margin is again used to calculate `minPos` and `maxPos` [https://github.com/emeeks/semiotic/blob/master/src/components/AnnotationLayer.js#L219](AnnotationLayer.js)

The effect is the labels appear to be "shifted down".

![marginalia_issue_screenshot](https://user-images.githubusercontent.com/6054688/50752354-d0926380-1202-11e9-9cab-2285b8b552bb.jpg)

The quick fix (at least for my use case) is to remove all the margins from the position calcs in AnnotationLayer.js. Otherwise, if the margins are intended for users to add custom margins specifically for the annotations, and are independent from the frame margins, then maybe those margins can be specified inside the layout object?

Either way, I can create a PR.

**Comment by emeeks** [comment]
Thanks for catching this. Feel free to open a PR otherwise I'll take a look at it.

**Comment by torioLuz** [comment]
As I looked more into the code, I realized this is the intended behavior. It does make sense to use up the margin space if real estate is limited and there are many labels. However, if number of labels are manageable (as in my example), I still want to see it not cross the axis boundaries. 

I've created a PR to allow a new layout option to override the margins. It shouldn't affect the default behavior. If this looks OK, I can update the docs. Otherwise, let me know if there's a more suitable approach.
